templates/settings.html

```diff
@@ -24,16 +24,52 @@ <h1 class="md:col-span-3 text-3xl break-words md:text-4xl font-bold text-slate-9
       {% /alert %}
     {% endif %}
 
-    <form method="POST" class="md:col-span-2 mt-6">
+    <form method="POST" class="md:col-span-2">
       {% csrf_token %}
 
       {% #card title="Your account" class="w-full" container_class="flex flex-col gap-y-4 items-start" container=True %}
         {% form_input autocapitalize="words" autocomplete="name" autocorrect="on" class="min-w-full" field=form.fullname label="Full name" required=True type="text" %}
-        {% form_input autocapitalize="off" autocomplete="email" autocorrect="off" class="min-w-full" field=form.notifications_email inputmode="email" required=True type="email" %}
-        {% #button type="submit" variant="primary" %}
+        {% form_input autocapitalize="off" autocomplete="email" autocorrect="off" class="min-w-full" field=form.notifications_email label="Contact Email" inputmode="email" required=True type="email" %}
+        {% #button type="submit" name="settings" variant="primary" %}
           Save your settings
         {% /button %}
       {% /card %}
     </form>
+
+    {% if request.user.social_auth.exists %}
+      {% #card title="Generate a Single Use Token" container=True class="md:col-span-2" %}
+        <div class="prose mb-6">
+          {% if token %}
+            <p>Your generated Single Use Token is:</p>
+            <div class="not-prose">
+              {% #code class="inline-flex !text-3xl font-bold !py-4 !px-8 rounded-md" %}
+                {{ token }}
+              {% /code %}
+            </div>
+          {% endif %}
+          <p>
+            You can use this to login to OpenSAFELY from within secure Level 4 environments.
+          </p>
+          <p>A Single Use Token is valid for logging in for:</p>
+          <ul>
+            <li>one login attempt, or</li>
```

Comment:
This might be a clearer way of phrasing the requirements (people can get confused with the use of logical-or)

> A Single Use Token is only valid:
> * for one login attempt
> * for one hour from the time you generate it
> * until you generate a new Single Use Token

Reply:
done